core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -604,6 +604,27 @@ public void testCreateNamespace() {
     // Test with no metadata
     catalog.createNamespace(testNamespace);
     assertThat(catalog.namespaceExists(testNamespace)).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb", "ns1"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb", "ns1", "ns2"))).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testD_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb."))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb.ns"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb.ns_"))).isFalse();
+    assertThat(catalog.namespaceExists(Namespace.of("testDb%"))).isFalse();
+
+    Namespace underscore = Namespace.of("te%t_b");
+    assertThat(catalog.namespaceExists(underscore))
+        .as("Should false to namespace doesn't exist")
+        .isFalse();
+
+    Namespace underscore2 = Namespace.of("special_name%space");
+    catalog.createNamespace(underscore2);
+    assertThat(catalog.namespaceExists(underscore2)).isTrue();
+    assertThat(catalog.namespaceExists(Namespace.of("special_name%spa_e")))
```

Comment:
For all of these test cases, can you add both positive and negative assertions? That is, test that the namespace doesn't exist and then create it and test that it does.

I think we also need to test existence by creating a nested namespace. That's probably a new test case.

Reply:
extended tests

Reply:
Looking good, but these are no longer tests of `createNamespace` and the test case is too long. Can you separate these into individual test cases for special characters?

Reply:
re organized tests, spitted tests per special character